### DISCUSSION TITLE
Keeping k3s-killall.sh up-to-date

### DISCUSSION THREAD

**Discussion by webD97** [k3s-io/k3s#11064] [General]
system-upgrade-controller and k3s-upgrade make it really easy to keep k3s up-to-date but they do not seem to manage `k3s-killall.sh` and `k3s-uninstall.sh`. I have a few machines that were installed ~ 2 years ago and they happily updated to k3s 1.30 over time but the outdated version of `k3s-killall.sh` there just bricked some clusters. Are there any plans to also update them together with k3s or was this a deliberate choice?

**Comment by brandond** [comment]
No, we don't have standalone versions of those currently available, and hadn't been planning to have them upgraded by the k3s-upgrade image, although we could consider it.

Can you point at what specifically broke your cluster when you used the old version?

**Reply by webD97** [reply]
Our old version of `k3s-killall.sh` (don't know how old exactly) does for example not include this commit: https://github.com/k3s-io/k3s/commit/3aceb85c22cc82ad2038898faebb04942e63ee87#diff-043df5bdbf6639d7a77e1d44c5226fd7371e5259a1e4df3a0dd5d64c30dca44fL846

Luckily, the affected clusters were only testing environments. We called the old `k3s-killall.sh` which then unmounted `/var/lib/rancher/k3s/storage` because we use a separate partition there. Then we restarted k3s. Since we do not have an explicit dependency between k3s and this mountpoint, we started without any PV data and broke both application as well as infrastructure components like monitoring.

We will patch our `k3s.service` to ensure the proper dependency on the mount point but imo it makes sense to keep these scripts up-to-date. If it had been, then the partition would not have been unmounted in the first place.
